apps/remote_control/lib/lexical/remote_control/code_intelligence/definition.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.Definition do
+  alias Lexical.SourceFile
+  alias Lexical.SourceFile.Position
+  alias Lexical.SourceFile.Range
+
+  require Logger
+
+  def definition(%SourceFile{} = source_file, %Position{} = position) do
+    maybe_location =
+      source_file
+      |> SourceFile.to_string()
+      |> ElixirSense.definition(position.line + 1, position.character + 1)
+
+    parse_location(maybe_location, source_file)
+  end
+
+  defp parse_location(%ElixirSense.Location{} = location, source_file) do
+    %{file: file, line: line, column: column} = location
+    file_path = file || source_file.path
+
+    case location_to_range(file_path, line, column) do
+      {:ok, {source_file, range}} ->
+        {:ok, %{source_file: source_file, range: range}}
+
+      error ->
+        error
+    end
+  end
+
+  defp parse_location(nil, _) do
+    {:ok, nil}
+  end
+
+  defp location_to_range(file_path, line, column) do
+    uri = SourceFile.Path.ensure_uri(file_path)
+
+    with {:ok, source_file} <- SourceFile.Store.open_temporary(uri),
+         {:ok, {:line, text, _, _, _}} <- SourceFile.fetch_line_at(source_file, line - 1) do
+      {:ok, {source_file, to_persisted_range(text, line, column)}}
+    else
+      _ ->
+        {:error, "Could not open source file: #{inspect(file_path)}"}
+    end
+  end
+
+  defp to_persisted_range(text, line, column) do
+    case Code.Fragment.surround_context(text, {line, column}) do
+      %{begin: begin, end: end_} ->
+        {begin_line, begin_column} = begin
+        {end_line, end_column} = end_
+        to_range({begin_line, begin_column}, {end_line, end_column})
```

Comment:
you're unpacking two tuples just to recreate them. 
```suggestion
      %{begin: start_pos, end: end_pos} ->
        to_range(start_pos, end_pos)
```